pkg/api/pluginproxy/token_provider_azure.go

```diff
@@ -44,9 +44,15 @@ func getAzureCredentials(cfg *setting.Cfg, authParams *plugins.JwtTokenAuth) azc
 	// * If authType and other fields aren't set then it means the datasource never been configured
 	//   and managed identity is the default authentication choice as long as managed identities are enabled
 	isManagedIdentity := authType == "msi" || (authType == "" && clientId == "" && cfg.Azure.ManagedIdentityEnabled)
+	isUserIdentity := authType == "userid" && cfg.Azure.UserIdentityEnabled
```

Comment:
If credentials explicitly configured as "userid" then they should not fallback to another authentication method even if `cfg.Azure.UserIdentityEnabled` is not enabled.

It's responsiblity of AzureTokenProvider to check whether user identity auth is supported (i.e. `cfg.Azure.UserIdentityEnabled` is enabled and properly configured).